pkg/tiller/server.go

```diff
@@ -87,3 +149,146 @@ func checkClientVersion(ctx context.Context) error {
 	}
 	return nil
 }
+
+func checkBasicAuth(ctx *context.Context) error {
+	md, ok := metadata.FromContext(*ctx)
+	if ok {
+		tokens, ok := md["authorization"]
+		if !ok {
+			return errors.New("Token not found")
+		}
+		apiserver, ok := md["k8s-server"]
+		if !ok {
+			return errors.New("Server url not found")
+		}
+		basicAuth, err := base64.StdEncoding.DecodeString(strings.TrimPrefix(tokens[0], "Basic "))
+		if err != nil {
+			return err
+		}
+		user, password := getUserPasswordFromBasicAuth(string(basicAuth))
+		if len(user) == 0 || len(password) == 0 {
+			return errors.New("Token not found")
+		}
+		kubeConfig := &rest.Config{
+			Host:     apiserver[0],
+			Username: user,
+			Password: password,
+		}
+		caData, ok := md["k8s-certificate-authority"]
+		if !ok {
```

Comment:
If `k8s-certificate-authority` is not found why not try with public cert or insecure?? @TamalSaha